README.md

```diff
@@ -375,6 +405,853 @@ Hull.logger.add(winstonSlacker,  { ... });
+    > Hostname of the current request. Since the connector are stateless services this information allows the connector to know it's public address.
+
+## Operations - configuration resolve strategy
+
+Here is what happens when your Express app receives a query.
+
+1. If a config object is found in `req.hull.config` steps **2** and **3** are skipped. 
+2. If a token is present in `req.hull.token`, the middleware will try to use the `hostSecret` to decrypt it and set `req.hull.config`.
+3. If the query string (`req.query`) contains `id`, `secret`, `organization`, they will be stored in `req.hull.config`.
+4. After this, if a valid configuration is available in `req.hull.config`, a Hull Client instance will be created and stored in `req.hull.client`.
+5. When this is done, then the Ship will be fetched and stored in `req.hull.ship`
+    > If there is a `req.hull.cache` registered in the Request Context Object, it will be used to cache the ship object. For more details see [Context Object Documentation](#context)
+6. If the configuration or the secret is invalid, an error will be thrown that you can catch using express error handlers.
+
+---
+
+# Hull.Connector
+The connector is a simple HTTP application served from public address. It could be implemented in any way and in any technological stack unless it implements the same API:
+
+```js
+const app = express();
+app.get("/manifest.json", serveTheManifestJson);
+app.listen(port);
+```
+
+Yet to ease the connector development and to extract common code base the `hull-node` library comes with the **Hull.Connector** toolkit which simplify the process of building new connector by a set of helpers and utilities which follows the same convention.
+
+## Initialization
+```js
+import Hull from "hull";
+
+const connector = new Hull.Connector({
+    port: 1234, // port to start express app on
+    hostSecret: "secret",  // a secret generated random string used as a private key
+    segmentFilterSetting: "synchronized_segments" // name of the connector private setting which has information about filtered segments
+});
+```
+
+This is the instance of the `Connector` module which exposes a set of utilities which can be applied to the main [express](http://expressjs.com/) app. The utilities can be taken one-by-one and applied the the application manually or there are two helper method exposed which applies everything be default:
+
+## Setup Helpers
+Setup Helpers are two high-level methods exposed by initialized Connector instance to apply custom middlewares to the Express application. Those middlewares enrich the application with connector features.
+
+```js
+import express from "express";
+import Hull from "hull";
+
+const app = express();
+const connector = new Hull.Connector({ hostSecret });
+
+connector.setupApp(app); // apply connector related features to the application
+app.post("/fetch-all", (req, res) => {
+  res.end("ok");
+});
+connector.startApp(app, port); // internally calls app.listen
```

Comment:
https://github.com/hull/hull-node/blob/release/v0.11.3/src/connector/hull-connector.js#L70
Not using `port` -> Remove from README.